### PR TITLE
Improved Game Played request and callback handling

### PR DESCRIPTION
**Pull request by OleksiiButeskul**
- added util methods for reading data from ServiceMethod inner body entity

### CODE REVIEW

File: steam-client/src/main/java/com/avenga/steamclient/base/ClientMessageProtobuf.java [hunk 1]
```diff
@@ -143,9 +143,8 @@ public BodyType getBody() {
     @Override
     public void deserialize(byte[] data) {
         Objects.requireNonNull(data, "data wasn't provided");
-        BinaryReader reader = new BinaryReader(new ByteArrayInputStream(data));
 
-        try {
+        try (BinaryReader reader = new BinaryReader(new ByteArrayInputStream(data))) {
```
Comment (andyislegend) [inline steam-client/src/main/java/com/avenga/steamclient/base/ClientMessageProtobuf.java]: var reader?

File: steam-client/src/main/java/com/avenga/steamclient/base/Message.java [hunk 2]
```diff
@@ -147,32 +147,30 @@ public void setSourceJobID(JobID jobID) {
 
     @Override
     public byte[] serialize() {
-        ByteArrayOutputStream baos = new ByteArrayOutputStream(0);
-        try {
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream(0)) {
```
Comment (andyislegend) [inline steam-client/src/main/java/com/avenga/steamclient/base/Message.java]: var?

File: steam-client/src/main/java/com/avenga/steamclient/base/ClientMessageProtobuf.java (hunk 1 above)
Comment (OleksiiButeskul) [reply steam-client/src/main/java/com/avenga/steamclient/base/ClientMessageProtobuf.java]: Fixed

File: steam-client/src/main/java/com/avenga/steamclient/base/Message.java (hunk 2 above)
Comment (OleksiiButeskul) [reply steam-client/src/main/java/com/avenga/steamclient/base/Message.java]: Fixed

**Pull request merged**